.pipelines/ccf.yml

```diff
@@ -12,8 +12,11 @@ steps:
       git clone --single-branch -b ccf-${{ parameters.Version }} https://github.com/microsoft/CCF $(Pipeline.Workspace)/CCF
     displayName: Checkout CCF ${{ parameters.Version }}
 
-  # This commit is in the main branch of CCF. Once it gets backported to 2.x we can bump the CCF version and stop patching.
-  - script: curl -L https://github.com/microsoft/CCF/commit/3c7f7cbaf6b27cff0cae61c02bda4af7fb6fff92.diff | patch -p1
+  # These are commits from the main branch of CCF to fix ASAN builds, but which
```

Comment:
Seems unrelated to this PR.

Reply:
It is related. The `cchost.py` uses CCF's json logger (whereas we previously used the text logger only), and [at some point in that code](https://github.com/microsoft/CCF/blob/d0990ec76e86a8555f16d3354116623ef41b7334/include/ccf/ds/logger.h#L142) the enclave offset gets converted from a double to a size_t, which is UB if the value does not fit in a size_t (this includes any negatives).

So whereas with the text logger we only get some garbage values printed, with the JSON logger trips ubsan and makes the process crash. The second patch I added fixes that bug: https://github.com/microsoft/CCF/pull/3972

I could ask the CCF team to backport this to 2.x, but given that running ASAN in apps isn't really supported on that branch anyway, it seems unlikely.